core/src/command/mod.rs

```diff
@@ -17,33 +14,34 @@ use tauri::{Builder, Runtime};
 use url::Url;
 use walkdir::{DirEntry, WalkDir};
 
-#[tauri::command(async)]
+#[tauri::command]
 /// Create a `.link`
 async fn create_link(
     title: String,
-    desc: String,
+    desc: Option<String>,
     url: String,
     state: tauri::State<'_, Cli>,
 ) -> Result<(), String> {
     let url = match Url::parse(url.as_str()) {
         Ok(val) => val,
         Err(e) => return Err(e.to_string()),
     };
-    let mut link = arklib::link::Link::new(title, desc, url);
-    let name = format!("{}.link", link.format_name());
-    dbg!(&name);
-    link.write_to_path(PathBuf::from(format!("{}/{}", &state.path, name)), true)
-        .await;
-    sleep(Duration::from_millis(305));
+    let ressource = arklib::id::ResourceId::compute_bytes(url.as_ref().as_bytes())
+        .expect("Error compute ressource from url");
+    let domain = url.domain().expect("Url has no domain");
+    let path = format!("{}/{domain}-{}.link", &state.path, ressource.crc32);
+    let mut link = Link::new(url, title, desc);
+    link.write_to_path(&state.path, &path, true)
+        .await
+        .expect("Custom error type needed");
```

Comment:
The Result type could be used here, Arklib uses anyhow 
Probably better to have a function return of Result<(), anyhow::Error> rather than a string and then just propagating it to the frontend with `?`

Reply:
We can't return anyhow::Error since it is not serializable. I propose to define custom error that are serializable